static/src/javascripts/projects/commercial/modules/lotame-data-extract.js

```diff
@@ -36,7 +35,37 @@ const init = (): Promise<void> => {
                 return Promise.reject(Error('No LOTCC in window'));
             }
         })
-        .catch(errorHandler);
+        .catch(errorHandler)
+
+// and stores in in window.OzoneLotameData
+const init = (): Promise<void> => {
+    if (!shouldLoadLotame || commercialFeatures.isSecureContact) {
+        return Promise.resolve();
+    }
+    if (shouldUseSourcepointCmp()) {
+        onConsentChange(state => {
+            let canRun: boolean = false;
+            if (state.tcfv2) {
+                // TCFv2 mode
+                if (
+                    typeof state.tcfv2.vendorConsents[SOURCEPOINT_ID] !== 'undefined'
```

Comment:
```suggestion
                    typeof state.tcfv2.vendorConsents !== 'undefined'
```

Reply:
I left that in to catch the case which there is no vendorConsent for that sourcepoint id. Having this check we ensure that we will always have a boolean instead of a potential `undefined` in the `canRun` and it will fallback to be checking all the purposes instead in such a scenario.